File: src/main/java/ro/nicolaemariusghergu/queryit/repository/ProductRepository.java
package ro.nicolaemariusghergu.queryit.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import org.springframework.transaction.annotation.Transactional;
import ro.nicolaemariusghergu.queryit.model.Product;

import java.util.List;
import java.util.Optional;

@Repository
@Transactional
public interface ProductRepository extends JpaRepository<Product, Long> {

    List<Product> findAllByCategoryId(Long id);

    Optional<Product> findByName(String name);
}

File: src/main/kotlin/ro/nicolaemariusghergu/queryit/repository/ProductRepository.kt
package ro.nicolaemariusghergu.queryit.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import org.springframework.transaction.annotation.Transactional
import ro.nicolaemariusghergu.queryit.model.Product

@Transactional
interface ProductRepository : JpaRepository<Product, Long>